sqlmesh/core/context.py

```diff
@@ -346,6 +346,7 @@ def __init__(
         self._metrics: UniqueKeyDict[str, Metric] = UniqueKeyDict("metrics")
         self._jinja_macros = JinjaMacroRegistry()
         self._requirements: t.Dict[str, str] = {}
+        self._environment_statements: t.List[EnvironmentStatements] = []
```

Comment:
Why is this a list?

Reply:
ah, multi project. I get it. But then should we always have one `EnvironmentStatements` and merge `before_all` and `afer_all` lists instead?

Reply:
Ah, python_envs can be different. I get it too